src/GeneralStateTestsFiller/stCreate2/CREATE2_HighNonceDelegatecallFiller.yml

```diff
@@ -0,0 +1,588 @@
+    comment: "Delegate calls CREATE/CREATE2 from an account with max allowed nonce/max allowed nonce - 1."
+  env:
+    currentCoinbase: 2adc25665018aa1fe0e6bc666dac8fc2697ff9ba
+    currentDifficulty: '0x020000'
+    currentGasLimit: '89128960'
+    currentBaseFee: '10'
+    currentNumber: '1'
+    currentTimestamp: '1000'
+    previousHash: 5e20a0453cecd065ea59c37ac63e079ee08998b6045136a8ce6635c7912ec0b6
+
+  pre:
+    a94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: 1000000000
+      code: ''
+      nonce: 0
+      storage: {}
+
+    000000000000000000000000fffffffffffffffe:
+      balance: 0
+      code: |
+        :yul
+        {
+          let createtype    := calldataload(0)
+          let contextnonce  := sload(0xffff)
+
+          // initcode: { return(0, 1) }
+          mstore(0, 0x60016000f3)
+          let addr
+          if eq(createtype, 0) {
+            addr := create(0, sub(32, 5), 5)
+          }
+          if eq(createtype, 1) {
+            // We use the context nonce to mimic CREATE's nonce based address calculation and make verification easier
+            addr := create2(0, sub(32, 5), 5, contextnonce)
+          }
+          sstore(2, addr)
+          if gt(addr, 0) { sstore(0xffff, add(contextnonce, 1)) }
+          mstore(0, addr)
+          return(0, 32)
+        }
+      nonce: '0xfffffffffffffffe'
+      storage: {
+        '0xffff': '0xfffffffffffffffe'
+      }
+
+    000000000000000000000000ffffffffffffffff:
+      balance: 0
+      code: |
+        :yul
+        {
+          let createtype    := calldataload(0)
+          let contextnonce  := sload(0xffff)
+
+          // initcode: { return(0, 1) }
+          mstore(0, 0x60016000f3)
+          let addr
+          if eq(createtype, 0) {
+            addr := create(0, sub(32, 5), 5)
+          }
+          if eq(createtype, 1) {
+            // We use the context nonce to mimic CREATE's nonce based address calculation and make verification easier
+            addr := create2(0, sub(32, 5), 5, contextnonce)
+          }
+          sstore(2, addr)
+          if gt(addr, 0) { sstore(0xffff, add(contextnonce, 1)) }
+          mstore(0, addr)
+          return(0, 32)
+        }
+      nonce: '0xffffffffffffffff'
+      storage: {
+        '0xffff': '0xffffffffffffffff'
+      }
+
+    b94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: 0
+      code: |
+        :yul
+        {
+          let calltype      := calldataload(4)
+          let callernonce   := calldataload(36)
+          let destnonce     := calldataload(68)
+          let createtype    := calldataload(100)
+
+          for { let contextnonce := sload(0xffff) } lt(contextnonce, callernonce) { contextnonce := sload(0xffff) } {
+            // We have a lower nonce than required for the caller, create dummy contract to increase nonce
+            mstore(0, 0x60016000f3)
+            let addr := create(0, sub(32, 5), 5)
+            if gt(addr, 0) { sstore(0xffff, add(contextnonce, 1)) }
```

Comment:
Else?

Reply:
I think else might be unnecessary here, since the store value is supposed to mimic the nonce and the nonce is not increased if the address returned is zero (contract creation failed).